### PR TITLE
Map compute specialization constants

### PR DESCRIPTION
**Pull request by grovesNL**
Shares code between graphics/compute pipelines setup to map compute specialization constants

### CODE REVIEW

File: libportability-gfx/src/conv.rs [hunk 1]
```diff
@@ -720,3 +720,43 @@ pub fn map_pipeline_statistics(flags: VkQueryPipelineStatisticFlags) -> query::P
...
+
+    entries
+        .into_iter()
+        .map(|entry| {
+            let offset = entry.offset as usize;
+            pso::Specialization {
+                id: entry.constantID,
+                value: match entry.size {
+                    4 => pso::Constant::U32(
+                        data[offset] as u32 |
```
Comment (kvark) [inline libportability-gfx/src/conv.rs]: nit: could probably look cleaner with a `fold`

File: libportability-gfx/src/conv.rs [hunk 2]
```diff
@@ -720,3 +720,43 @@ pub fn map_pipeline_statistics(flags: VkQueryPipelineStatisticFlags) -> query::P
...
+                        (data[offset+4] as u64) << 32 |
+                        (data[offset+5] as u64) << 40 |
+                        (data[offset+6] as u64) << 48 |
+                        (data[offset+7] as u64) << 56
+                    ),
+                    size => panic!("Unexpected specialization constant size: {:?}", size),
+                },
+            }
+        })
+        .collect::<Vec<pso::Specialization>>()
```
Comment (kvark) [inline libportability-gfx/src/conv.rs]: nit: the explicit type is not needed here

File: libportability-gfx/src/conv.rs (hunk 1 above)
Comment (kvark) [reply libportability-gfx/src/conv.rs]: @msiglreith would probably argue that it doesn't worth the risk of overhead... and that would be correct.
I was thinking about something like this:
```rust
data[offset..offset+4].iter().rev().fold(0u32, |u, &v| (u << 8) | v)
```

Comment (grovesNL) [reply libportability-gfx/src/conv.rs]: Ah ok, so should I just leave this as it is for now?

**Pull request merged**